packages/midi-input/src/MidiInputPanel.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from 'react'
+import { EnumOption, HedronEngine, Input } from '@hedron/engine'
+import {
+  Button,
+  ControlGrid,
+  EnumDropdown,
+  NodeControl,
+  NodeControlInner,
+  NodeControlMain,
+  NodeControlTitle,
+} from '@hedron/ui-core'
+import { MidiInput, MidiInputOptions } from './MidiInput'
+
+type Entries<T> = {
+  [K in keyof T]-?: [K, T[K]]
+}[keyof T][]
+interface IProps {
+  input: Input<MidiInputOptions>
+  // TODO: This can be typed as something like HedronEngineWithPlugin<MidiInput>
+  engine: HedronEngine
+}
+
+const channelOptions: EnumOption[] = []
+for (let i = 0; i <= 15; i++) {
+  channelOptions.push({ label: `${i + 1}`, value: i })
+}
+
+const noteOptions: EnumOption[] = []
+for (let i = 0; i < 127; i++) {
+  noteOptions.push({ label: `${i + 1}`, value: i })
+}
+
+const enumOptions = {
+  channel: channelOptions,
+  note: noteOptions,
+}
+
+/**
+ * A react component that displays the midi settings for a parameter
+ */
+export const MidiInputPanel = ({ input, engine }: IProps) => {
+  const [isLearning, setIsLearning] = useState(false)
+
+  // TODO: May not need this "as" if we have HedronEngineWithPlugin<MidiInput>
+  const plugin = engine.plugins['midi-input'] as MidiInput
+  const midi = plugin.midiManager
```

Comment:
The component has access to whatever functionality it needs from the plugin by getting it from the engine. (e.g. `engine.plugins['my-plugin-id'].myCoolThing`)